### ISSUE TITLE
ULPM doesn't work with MCU version 002.011 for wp77xx

### ISSUE THREAD

**Issue by akankshaashwini** [legatoproject/legato-af#45]
Tested with pmtool on a Mangoh red with wp77xx module.
> pmtool bootOn timer 60
> SUCCESS!
> pmtool shutdown
> Initiated the MDM shutdown
Doesn't perform a shutdown

**Comment by CoRfr** [comment]
Which version of Legato did you test? Also which Linux

**Comment by akankshaashwini** [comment]
I tested with legato 19.04 version. Linux 3.18. Toolchain SWI9X06Y_02.22.12.00.
I used Release 11 resources just with an updated legato version to 19.04:
https://source.sierrawireless.com/resources/airprime/software/wp77xx/wp77xx-firmware-latest-release-components/

**Comment by jvermillard** [comment]
if you have the legato devMode app, I don't know why, but it's preventing ULPM to shutdown the module. Try to remove it.

**Comment by akankshaashwini** [comment]
I don't find the devMode app installed.

**Comment by CoRfr** [comment]
Yes the `devMode` app keeps the target alive. See https://docs.legato.io/latest/basicTargetDevMode.html:
> - holds a wake lock, so the target won't go to sleep when disconnected from USB.

I created an internal tracker, LE-13385, will keep you posted.

**Comment by Redferne** [comment]
Can confirm this problem with the same setup (WP7702), also tried Legato 19.05. However it seems that PSM can still be triggered by using the psm_test command and the system powers down.

Can also report that kernel autosleep is not working when legato is running (maybe this is intended?) Is there a workaround?

I'm connected to the console UART port (not CF3 USB) and devMode is not enabled.

**Comment by auandrew** [comment]
I cannot reproduce this issue with all the exact same versions. 

> root@swi-mdm9x28-wp:~# microcom /dev/ttyAT
> 
> ati8
> Legato Ver: 19.04.0_a62fc5efed553c95ae8e2af417420976
> Yocto Ver: SWI9X06Y_02.22.12.00 2019-04-24_20:46:59
> OS Ver: Linux version 3.18.44 (jenkins@jenkins) (gcc version 6.2.0 (GCC) ) #2 PREEMPT Wed Apr 24 20:48:03 UTC 2019
> LK Ver: SWI9X06Y_02.16.02.00
> RootFS Ver: SWI9X06Y_02.22.12.00 2019-04-24_20:46:59
> UserFS Ver: unknown
> MCU Ver: 002.011

In general, any active wake lock would prevent Legato's ULPM from proceeding with a shutdown, e.g. 

> Jan  6 01:23:16 swi-mdm9x28-wp user.info Legato:  INFO | powerManagerService[2899]/powerMgr T=main | le_pm.c OnClientConnect() 270 | Connection from client pmtool/3195
> Jan  6 01:23:16 swi-mdm9x28-wp user.err Legato: =ERR= | powerManagerService[2899]/powerMgr T=main | le_ulpm.c le_ulpm_ShutDown() 499 | Wakelock held!! System can't be shut down. Try again.
> Jan  6 01:23:16 swi-mdm9x28-wp user.info Legato:  INFO | powerManagerService[2899]/powerMgr T=main | le_pm.c OnClientDisconnect() 298 | Client pid 3195 disconnected.

Thus, in your environment where the issue manifests, is there any other active wake, just like devMode but not devMode itself? 

There is related documentation about wake locks at
- https://docs.legato.io/latest/conceptsPwrMgmt.html 

and pmtool at 
- https://docs.legato.io/latest/toolsTarget_pmTool.html.
